backup/providers.go

```diff
@@ -6,16 +6,22 @@ import (
 	"encoding/json"
 	"fmt"
 
+	filestore "github.com/featureform/filestore"
 	"github.com/featureform/provider"
 	pc "github.com/featureform/provider/provider_config"
 	"gocloud.dev/gcp"
 )
 
 type Provider interface {
 	Init() error
-	Upload(name, dest string) error
-	Download(src, dest string) error
-	LatestBackupName(prefix string) (string, error)
+	// **NOTE:** Upload accepts strings instead of filestore.Filepath because
+	// the store, which could create a filestore-specific filepath, is not exposed
+	// by the Provider interface, and it seems unnecessary to expose it at this time.
+	// In contrast, Download accepts a filestore.Filepath because the LatestBackupName
+	// returns a filestore.Filepath and dest is an instance of filestore.LocalFilepath.
+	Upload(src, dest string) error
+	Download(src, dest filestore.Filepath) error
+	LatestBackupName(filenamePrefix string) (filestore.Filepath, error)
```

Comment:
I know there wasn't a comment before but it might be useful to know what this is